notebooks/dataAccess/siphon-RemoteOpen.ipynb

```diff
@@ -0,0 +1,378 @@
+   "outputs": [],
+   "source": [
+    "data_file = ds.remote_open()\n",
+    "data_file"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now have a object that we can read similar to a local file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = data_file.readline()\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*Note:* When we use `remote_open` to read a dataset, we are reading raw data from a file-like object, rather than formatted data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"#top\">Top</a>\n",
+    "\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Read the returned object like a local file\n",
+    "We can now read our dataset using random access."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can read a line, as we did in the previous section, or we can read a specified number of bytes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data = data_file.read(100)\n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can change our position in the file using `seek`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_file.seek(0) # go to start of file\n",
+    "print(data_file.read(4)) # print first 4 bytes\n",
+    "data_file.seek(50) # go to position 50\n",
+    "print(data_file.read(10)) # print 10 more bytes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And we can read the data directly into a byte array."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "b = bytearray(100) # create a byte array of length 100\n",
+    "data_file.readinto(b) # read 100 bytes into the byte array\n",
+    "b[:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Calling `getbuffer` returns the location in memory where the dataset is being stored locally."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "b = data_file.getbuffer()\n",
+    "b"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use the memory buffer to make local writes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_file.seek(100) # go to position 100\n",
+    "b[100:110] = b\"helloworld\"; # write 'helloworld' into our local data\n",
```

Comment:
I think you don't want the 'b' in this line next to "helloworld"

Reply:
the b is there to signify that it's bytes, otherwise it won't accept the string. I will make that clear with a comment though, cuz I didn't know that  at first either and I bet a bunch of students wouldn't.

Reply:
Okay, cool. I think I might also have you add a quick not in section 2, because that's the first time you print out the data, and when it prints out it starts with 

> b'GRIB.....

And so I'm assuming that 'b' is also there to signify bytes.  But might be good to mention it to the user at this point in the notebook as well?

Reply:
Sure, that sounds good. One sec.